template/src/hooks/useQuery.ts

```diff
@@ -0,0 +1,26 @@
+import * as queryString from "query-string";
+import { useLocation } from "react-router";
+import { compact } from "lodash";
+
+// Parse the URLs query param into an object of type T.
+// query param keys that represent an array are provided in <arrayKeys>.
+//
+// E.g. useQuery<{name: string; titles: string[];}({arrayKeys: ["titles"]}); to
+// parse ?name=Max&titles=BA&titles=MA into { name: "Max", titles: ["BA", "MA"] }
+//
+// Default to void so we get a compile error, if users are not providing an explicit type. This avoids implicit any.
+export const useQuery = <T = void>(options?: { arrayKeys?: string[] }) => {
+    const location = useLocation();
+    const result = queryString.parse(location.search, { parseBooleans: true }) as any;
```

Comment:
useHash should probably also use `parseBooleans`, also I am missing the arrayFormat configuration